Re-write global header guidance (#1634)
grommet/hpe-design-system: aries-site/src/pages/templates/global-header.mdx

import {
  GlobalHeaderExample,
} from '../../examples';
import { Example } from '../../layouts';

> import { Box, Image } from 'grommet';
> import { AppsRounded, Language, Search, Shop } from 'grommet-icons';

Reviewer: ```suggestion
import { Box, , Button, Image } from 'grommet';
import { AppsRounded, Language, Search, Shop } from 'grommet-icons';

The Global Header, also known as the HPE Common Header and Footer Web Service, is a standardized header to be used across all HPE applications. The header is provided as a service and injected into your application at run-time.

<Button label="View Implementation Details" href="https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php" target="_blank" rel="noreferrer  noopener" secondary />

<Button label="Development Team Contact" href="mailto:hpecom-cross-platform-development@hpe.com?subject=Common%20Header-Footer%20Service" secondary />
```

Author: Sounds great.

<Example
  code={[
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/templates/global-header/components/GlobalHeaderExample.js',
  ]}
  width="100%"
  screenContainer
>
  <GlobalHeaderExample />
</Example>

## Guidelines

> The Global Header is a standardized header to be used across applications. HPE provides an implementation
> that you can reference from your application. You can choose various features to enable like country selector,
> shopping cart and HPE passport login, etc. You can also enable or disable the page footer with privacy statement
> link, terms of use, ad choices and cookies, and *Do Not Sell My Information* links.
> 
> Each site or application will have different needs. Some header options may not make sense for some
> sites. The following sections will help you decide which options to include.

Reviewer: ```suggestion
The Global Header is a standardized header to be used across applications. HPE [provides an implementation](https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php)
to be referenced from your application. 

Both the header and footer are customizable, allowing consumers to choose various features to enable such as country selector,
shopping cart, HPE passport login, privacy statement links, theme mode, and more.
Each site or application will have different needs. Some header options may not make sense for some
sites. The following sections will help you decide which options to include.
```

Author: Sounds great

of that page will change to include those options, and the code sample on that page will 
change to declare the chosen options as parameters in
`<script src="https://h50007.www5.hpe.com/hfws/us/en/hpe/latest.r/root?...">` element that
configures the header.

You can then copy the URL from that `<script>` and use it in the code sample at the top of
this page to configure the global header and footer for your site.

Let's take a closer look at each part of the global header and footer.

### App Identity

Most sites should choose the default which will be the HPE logo for regular mode or the GreenLake logo for
`console` mode. You can force it to use one or the other by specifically choosing HPE versus GreenLake if
needed.

![HPE logo](/templateImages/globalheader-hpe.png)

![GreenLake logo](/templateImages/globalheader-greenlake.png)

### Header Main Navigation

The main navigation in the center of the header provide marketing content. The links allow users to find
HPE products, contact support and get more information about HPE. These should also typically be enabled
for most sites.

![Global Header Main Navigation example](/templateImages/globalheader-nav.png)

### Search

> The *Search* icon allows the user to search hpe.com. 
> 
> <Search />

Reviewer: Present inline?

```suggestion
<Search /> The *Search* icon allows the user to search hpe.com. 
```

Author: Done

### Shopping Cart

The *Shopping Cart* icon goes to the HPE shopping cart for buy.hpe.com. This should only
be enabled for sites which have products or solutions for sale.

<Shop />

### Country Selector

The country selector allows the user to go to a language-specific version of hpe.com. This
should be disabled for most sites since the link really only takes you back to hpe.com
instead of staying on your site.

<Language />

### Sign in

Sign in allows the user to sign in to their HPE Passport account for hpe.com. Enable this
option if your site uses features that require a Passport login.

![Sign In button](/templateImages/globalheader-sign-in.png)

### Cube Menu

The *Cube Menu* allows access to HPE services available to the user such as GreenLake and
other Cloud console applications as well as general HPE links for HPE content like
developer.hpe.com, education.hpe.com, etc. This menu should be enabled to allow access
to other HPE sites and applications.

<AppsRounded />

> 

Reviewer: Include Color Scheme option?

Author: I wasn't sure what to do with that option. I think I'll check if there's some guidance we can provide for this choice.

Reviewer: This is the most documentation I've seen and it doesn't say much.
[common-hfws-quick-js-guide-1.0.0 (3).pdf](https://github.com/grommet/hpe-design-system/files/6593348/common-hfws-quick-js-guide-1.0.0.3.pdf)

What about:
"Color Scheme allows implementors to choose either a light or dark variant of the header/footer. The color scheme you choose should be harmonious with the primary background of your application's header and avoid stark contrast which interrupts a user's visual flow. If your application supports light and dark modes, consider setting the `color_scheme` parameter with a value of either `light` or `dark` in alignment with the user's preference."

To the last point, in our implementation of Global Header, we could reference the localStorage variable `darkMode` --> there is a `useDarkMode()` hook in utils.

Author: Thanks for the wording. I think that works. I like the useDarkMode hook but I don't think we should recommend using an undocumented grommet API.

Reviewer: > I don't think we should recommend using an undocumented grommet API.

Agree. Was meaning that on the Design System site we should change the Global Header color_scheme to match the themeMode someone has chosen for the site.

Reviewer: That would be an enhancement outside of this PR, it just happened to be an aside thought I had as I was writing the guidance for the page.

> ## What makes up the Global footer

Reviewer: ```suggestion
## What makes up the Global footer?
```

Author: Done

The global footer contains links for privacy information and various opt-out and choices required by
European GDPR standards as well US and other privacy requirements. In general these links should
be enabled for customer-facing sites.

### Privacy Link

The *Privacy* link goes to HPE's Privacy Statement which explains how HPE manages and protects a users
information and respects the relevant privacy standards like GDPR, APEC, etc.

### Terms of Use Link

The *Terms of Use* link goes to the legal statement that governs the user's activity on HPE's websites.

### Ad Choices and Cookies Link

The *Ad Choices and Cookies* link goes to the section of the *Privacy Statement* that explains how
cookies are used as well as how you can opt-out of targetted advertisements and analytics data
such as Google Analytics and Adobe Analytics.

### Do Not Sell My Personal Information Link

The *Do Not Sell My Personal Information* link goes to information on how to opt-out of any sale

> of personal collected by automatic data collection tools.

Reviewer: Maybe include links again for convenience? Soft opinion.

```suggestion
of personal collected by automatic data collection tools.

<Box direction="row" gap="medium">
    <Button label="View Implementation Details" href="https://h50007.www5.hpe.com/hfws-test-pages/common-hfws-test-page.php" target="_blank" rel="noreferrer  noopener" secondary />
    <Button label="Development Team Contact" href="mailto:hpecom-cross-platform-development@hpe.com?subject=Common%20Header-Footer%20Service" secondary />
</Box>
```

Author: done
